lib/IRGen/IRGenModule.cpp

```diff
@@ -392,7 +392,7 @@ IRGenModule::IRGenModule(IRGenerator &irgen,
     AtomicBoolAlign = Alignment(ClangASTContext->getTypeSize(atomicBoolTy));
   }
 
-  IsSwiftErrorInRegister =
+  IsSwiftErrorInRegister = UseSwiftCC &&
```

Comment:
Why was the additional `UseSwiftCC` needed again?

Reply:
To summarise my earlier answer: both ‘UseSwiftCC’ and ‘IsSwiftErrorInRegister’ are checked when deciding whether to emit the ‘swifterror’ attribute, but only the latter is checked later in the code, so there’s a mismatch - it expects there to be attributes present that weren’t added.

Reply:
This is the least disruptive way to fix that particular issue, but the correct solution may be to change one of the problematic functions to match - which to change depends on whether IsSwiftErrorInRegister should require UseSwiftCC or not.

Reply:
That being a workaround is reasonable, though we should at least know where exactly it shows up and how to reproduce it so that we can track fixing it.  I think it is worth filing an SR for that issue.

Reply:
So the exact locations are line 163 of `GenCall.cpp`, where it has

`if (!UseSwiftCC || !this->IsSwiftErrorInRegister)`

as opposed to line 2204 of `GenCall.cpp` and line 3369 of `IRGenSIL.cpp`, which check

`if (IGM.IsSwiftErrorInRegister)`.

The comments before lines 163 and 2204 of `GenCall.cpp` more-or-less match, so one of those conditions is incorrect; my guess would be that `IsSwiftErrorInRegister` should be dependent on `UseSwiftCC` all of the time, and then we're just left with a duplicated test in line 163 of `GenCall.cpp`.

I won't have a chance to try a native Windows build for a while and give you the exact error message, but that's the reasoning.